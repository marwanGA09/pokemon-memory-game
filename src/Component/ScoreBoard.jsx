import PropTypes from 'prop-types';

function ScoreBoard({ score, highScore }) {
  return (
    <div className="score-board">
      <h3>Memory Game</h3>
      <p>
        Score: <span>{score}</span>
      </p>
      <p>
        High Score: <span className="high-score">{highScore}</span>
      </p>
    </div>
  );
}

ScoreBoard.propTypes = {
  score: PropTypes.number,
  highScore: PropTypes.number,
};

export { ScoreBoard };
