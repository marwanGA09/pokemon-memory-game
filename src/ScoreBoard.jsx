export function ScoreBoard({ score, highScore }) {
  return (
    <div className="score-board">
      <h3>Memory Game</h3>
      <p>
        Score: <span>{score}</span>
      </p>
      <p>
        High Score: <span className="high-score">{highScore}</span>
      </p>
    </div>
  );
}
