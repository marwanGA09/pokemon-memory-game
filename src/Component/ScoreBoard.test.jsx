import { render } from '@testing-library/react';
import { describe, test, expect } from 'vitest';
import { ScoreBoard } from './ScoreBoard';

describe('test scoreBoard', () => {
  test('snapshot test', () => {
    const { asFragment } = render(<ScoreBoard />);
    expect(asFragment()).toMatchSnapshot();
  });
});
