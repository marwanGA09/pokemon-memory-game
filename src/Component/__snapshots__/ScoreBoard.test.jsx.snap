// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test scoreBoard > snapshot test 1`] = `
<DocumentFragment>
  <div
    class="score-board"
  >
    <h3>
      Memory Game
    </h3>
    <p>
      Score: 
      <span />
    </p>
    <p>
      High Score: 
      <span
        class="high-score"
      />
    </p>
  </div>
</DocumentFragment>
`;
